src/components/Table.tsx

```diff
@@ -166,15 +173,15 @@ export const Table = <TableRowData extends object, TableRowKey extends Key>({
   ) => {
     if (!sortColumn) return 0;
 
-    const column = columns.find((column) => column.columnKey === sortColumn);
+    const column = columns.find(({ columnKey }) => columnKey === sortColumn);
     const first = column?.getCellValue(a);
     const second = column?.getCellValue(b);
 
     return (
       // Compare the items by the sorted column
-      (isNaN(first as number)
+      (Number.isNaN(Number(first))
         ? // String
-          collator.compare(first as string, second as string)
+          collator.compare(String(first), String(second))
```

Comment:
i feel like we could use `typeof first === 'string' && typeof second === 'string'` instead of typescasting here

Reply:
i think it depends on the intended effect - do we want to be able to sort `'1'` with `2`? that's the current functionality which i'm preserving. (this should never happen though?)

i do agree your way is more correct though. how do you feel about doing that as a followup to this PR? so we can isolate that PR for regression tracking